src/utils.ts

```diff
@@ -0,0 +1,111 @@
+import * as vscode from 'vscode'
+import * as temp from 'temp'
+import * as fs from 'fs'
+import { isArray } from 'util'
+
+export function sanitizedDateString(date?: Date) {
+    const d = date || new Date();
+    const pad = (num: number) => ("00" + num).slice(-2);
+
+    return `${d.getFullYear()}-${pad(d.getMonth())}-${pad(d.getDay())}-${pad(d.getHours())}-${pad(d.getMinutes())}-${pad(d.getSeconds())}`;
```

Comment:
Not one for formatting it in a file-name-friendly way AFAIK. I've seen a fair number of custom packages to format dates which makes me think that there isn't a builtin for it.